### PR TITLE
Fix telemetry reporter logging

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Backport fixes from 1.2.1 and improve tests.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — DISMISSED** [review]
:shipit:

**Review by noisersup — DISMISSED** [review]
Nice catch!

**Review by w84thesun — DISMISSED** [review]
LGTM

**Comment by AlekSi** [comment]
There is another thing to improve, but not in a rush

**Review by chilagrow — APPROVED** [review]

**Review by noisersup — APPROVED** [review]

File: internal/util/telemetry/reporter_test.go [hunk 1]
```diff
@@ -94,39 +94,40 @@ func TestNewReporterLock(t *testing.T) {
...
+func TestReporterReport(t *testing.T) {
+	t.Parallel()
 
 	t.Run("TelemetryEnabled", func(t *testing.T) {
-		serverCalled = 0 // telemetry server was not called yet.
+		t.Parallel()
+
+		var serverCalled int
+		telemetryResponse := response{
+			LatestVersion:   "v1.2.1",
```
Comment (chilagrow) [inline internal/util/telemetry/reporter_test.go:121]: Thanks for updating this too!

**Review by chilagrow — APPROVED** [review]

**Pull request merged**